RavenDB-11893

aviv86 opened the pull request:
http://issues.hibernatingrhinos.com/issue/RavenDB-11893

aviv86 committed:
RavenDB-11893 : PatchDocumentCommand should clear the UpdateCountersInMetadata hashset from it's ScriptRunner.SingleRun after each run

ayende commented on src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs:
```diff
@@ -186,6 +186,8 @@ public override int Execute(DocumentsOperationContext context)
                         }
                     }
+
+                    _run.UpdatedDocumentCounterIds.Clear();
```
Did this mean that we were updating _all_ the counters for all the previous documents on each operation?

aviv86 replied:
Yes

ppekrol commented on src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs:
```diff
@@ -186,6 +186,8 @@ public override int Execute(DocumentsOperationContext context)
                         }
                     }
+
+                    _run.UpdatedDocumentCounterIds.Clear();
```
This should go to `SingleRun.Reset` probably?

ppekrol committed:
RavenDB-11893
- moved UpdatedDocumentCounterIds.Clear to SingleRun.Reset and ReturnRun.Dispose
- cleanup
